File: sirius-core/src/main/java/org/starcoin/util/WaitUtil.java
package org.starcoin.util;

import java.util.concurrent.TimeUnit;
import java.util.concurrent.atomic.AtomicBoolean;
import java.util.function.BooleanSupplier;

public class WaitUtil {


    public static void waitFor(AtomicBoolean stop, long waitInterval, TimeUnit timeUnit,
                               BooleanSupplier condition) {
        while (!condition.getAsBoolean() && !stop.get()) {
            wait(waitInterval, timeUnit);
        }
    }

    public static void wait(long waitTime, TimeUnit timeUnit) {
        try {
            Thread.sleep(timeUnit.toMillis(waitTime));
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
            throw new RuntimeException(e);
        }
    }
}

File: sirius-core/src/main/kotlin/org/starcoin/sirius/util/WaitUtil.kt
package org.starcoin.sirius.util

import java.util.concurrent.TimeUnit
import java.util.concurrent.atomic.AtomicBoolean
import java.util.function.BooleanSupplier

object WaitUtil {


    fun waitFor(
        stop: AtomicBoolean, waitInterval: Long, timeUnit: TimeUnit,
        condition: BooleanSupplier
    ) {
        while (!condition.asBoolean && !stop.get()) {
            wait(waitInterval, timeUnit)
        }
    }

    fun wait(waitTime: Long, timeUnit: TimeUnit) {
        try {
            Thread.sleep(timeUnit.toMillis(waitTime))
        } catch (e: InterruptedException) {
            Thread.currentThread().interrupt()
            throw RuntimeException(e)
        }

    }
}
